### PR TITLE
Change windowHandle to windowId when EVENT_DESTROY_WINDOW

### PR DESCRIPTION
**Pull request by freeboy**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/bindings/event/EventDispatcher.cpp [hunk 1]
```diff
@@ -423,17 +422,15 @@ void EventDispatcher::dispatchDestroyWindowEvent(cc::ISystemWindow *window) {
...
 #if CC_PLATFORM == CC_PLATFORM_WINDOWS
-    EventDispatcher::dispatchCustomEvent(EVENT_RECREATE_WINDOW, 1,
-                                         reinterpret_cast<void *>(CC_GET_MAIN_SYSTEM_WINDOW()->getWindowHandle()));
+    EventDispatcher::dispatchCustomEvent(EVENT_RECREATE_WINDOW, 1, ISystemWindow::mainWindowId);
 #else
     EventDispatcher::dispatchCustomEvent(EVENT_RECREATE_WINDOW, 0);
 #endif
 }
 
 void EventDispatcher::dispatchRecreateWindowEvent(cc::ISystemWindow *window) {
```
Comment (bofeng-song) [inline native/cocos/bindings/event/EventDispatcher.cpp:431]: Whether it is possible to consider merging the two versions of dispatchRecreateWindowEvent

Comment (freeboy) [reply native/cocos/bindings/event/EventDispatcher.cpp:431]: It should be possible, the previous overloading was for compatibility.

**Comment by freeboy** [comment]
@cocos-robot run test cases

**Comment by freeboy** [comment]
@cocos-robot run test cases

**Review by minggo — APPROVED** [review]

**Review by bofeng-song — APPROVED** [review]

**Review by bluesky013 — APPROVED** [review]

**Comment by freeboy** [comment]
@cocos-robot run test cases

**Comment by pandamicro** [comment]
https://github.com/cocos/cocos-engine/issues/10893

**Pull request merged**